packages/commerce-sdk-react/src/test-utils.tsx

```diff
@@ -110,7 +110,7 @@ export const mockMutationEndpoints = (matchingPath: string, options?: {errorResp
         .delete((uri) => {
             return uri.includes(matchingPath)
         })
-        .reply(options?.errorResponse ? options.errorResponse : 204)
+        .reply(options?.errorResponse ? options.errorResponse : 204, NEW_DATA)
```

Comment:
This makes me realize that some API endpoints that receive DELETE requests would respond in 1 of 2 ways:
- either return 200 → for example, [this](https://developer.salesforce.com/docs/commerce/commerce-api/references/shopper-baskets?meta=removeCouponFromBasket) and [that](https://developer.salesforce.com/docs/commerce/commerce-api/references/shopper-orders?meta=removePaymentInstrumentFromOrder) endpoints
- or 204 (which means successful but the response has empty body) → for example, [this](https://developer.salesforce.com/docs/commerce/commerce-api/references/shopper-customers?meta=removeCustomerAddress) and [that](https://developer.salesforce.com/docs/commerce/commerce-api/references/shopper-baskets?meta=deleteBasket) endpoints

Reply:
I think for testing purpose, we can simply choose 1 of the above. Say, since currently the mocked response returns 204, then it should not return anything.. no need for `NEW_DATA`.

Reply:
Gotcha. I am going to change this to 200 and return `NEW_DATA` because for testing purposes, if we return an empty response body, line 121 in the test-utils file fails: `expect(queryResult.data).toEqual(newData)`